src/signature_with_sighash_info.rs

```diff
@@ -0,0 +1,127 @@
+//! Defines `SighashInfo` and signatures with `SighashInfo`.
```

Comment:
File name should be `sighash_version.rs`
Please extend doc a bit / fix names

Reply:
Done